src/dex/fluid-dex/fluid-dex-pool.ts

```diff
@@ -88,28 +214,27 @@ export class FluidDexEventPool extends StatefulEventSubscriber<FluidDexPoolState
   async generateState(
     blockNumber: number,
   ): Promise<DeepReadonly<FluidDexPoolState>> {
-    // TODO: complete me!
-    const poolReserves = await this.dexHelper.multiContract.methods
-      .getPoolReserves(this.pool.address)
-      .call({}, blockNumber);
-
-    return poolReserves;
-    // return {
-    //   token0RealReserves: poolReserves.collateralReserves.token0RealReserves.toString(),
-    //   token1RealReserves: poolReserves.collateralReserves.token1RealReserves.toString(),
-    //   token0ImaginaryReserves: poolReserves.collateralReserves.token0ImaginaryReserves.toString(),
-    //   token1ImaginaryReserves: poolReserves.collateralReserves.token1ImaginaryReserves.toString(),
-    //   token0Debt: poolReserves.debtReserves.token0Debt.toString(),
-    //   token1Debt: poolReserves.debtReserves.token1Debt.toString(),
-    // };
-  }
+    const ResolverAbi = new Interface(ResolverABI);
+    const callData: MultiCallParams<PoolWithReserves>[] = [
+      {
+        target: this.pool.resolver,
+        callData: ResolverAbi.encodeFunctionData('getPoolReserves', [
+          this.pool.address,
+        ]),
+        decodeFunction: await this.decodePoolWithReserves,
+      },
+    ];
 
-  // Its just a dummy example
-  handleMyEvent(
-    event: any,
-    state: DeepReadonly<FluidDexPoolState>,
-    log: Readonly<Log>,
-  ): DeepReadonly<FluidDexPoolState> | null {
-    return null;
+    const results: PoolWithReserves[] =
+      await this.dexHelper.multiWrapper.aggregate<PoolWithReserves>(
+        callData,
+        blockNumber,
+        this.dexHelper.multiWrapper.defaultBatchSize,
+      );
+
+    return {
```

Comment:
likely somewhere we need to track and calculate the fee amount? is this not needed for Paraswap? can you check in other projects code?

Reply:
now fetching from resolver itself, so maybe this is resolved